crates/types/src/schema/service.rs

```diff
@@ -210,6 +335,11 @@ pub trait ServiceMetadataResolver {
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct HandlerSchemas {
     pub target_meta: InvocationTargetMetadata,
+    pub idempotency_retention: Option<Duration>,
+    pub workflow_completion_retention: Option<Duration>,
```

Comment:
Still wondering whether this shouldn't be simply `retention` and then we have some system defaults for workflow and idempotency calls which are used if no other retention duration was configured. Maybe something for the future I guess.

Reply:
at this level we could do the merge already, that's true. But I want to keep a flat data model here similar to the deployment manifest